finetune.py

```diff
@@ -221,6 +223,180 @@ def build_train_loader(
+                if not line or line.startswith('#'):
+                    continue
+                
+                parts = line.split()
+                if len(parts) != 2:
+                    logging.warning(f"Skipping invalid line: {line}")
+                    continue
+                
+                element, energy = parts
+                try:
+                    # Try as atomic number
+                    z = int(element)
+                    if 1 <= z <= 118:
+                        ref_energies[z] = float(energy)
+                except ValueError:
+                    # Try as element symbol
+                    if element in ELEMENT_SYMBOLS:
+                        z = ELEMENT_SYMBOLS[element]
+                        ref_energies[z] = float(energy)
+                    else:
+                        logging.warning(f"Unknown element: {element}")
+        
+        logging.info(f"Loaded reference energies from text file: {filepath}")
+    
+    return ref_energies
+
+
+def configure_model(model, args, device):
+    """Configure model with custom loss weights and reference energies."""
+    is_conservative = isinstance(model, ConservativeForcefieldRegressor)
+    is_direct = isinstance(model, DirectForcefieldRegressor)
+    
+    logging.info(f"Model type: {type(model).__name__}")
+    logging.info(f"  Conservative: {is_conservative}")
+    logging.info(f"  Direct: {is_direct}")
+    
+    # Configure loss weights
+    if args.energy_loss_weight is not None or args.forces_loss_weight is not None or args.stress_loss_weight is not None:
+        logging.info("=" * 60)
+        logging.info("Configuring custom loss weights...")
+        
+        # Set energy weight
+        if args.energy_loss_weight is not None:
+            model.loss_weights["energy"] = args.energy_loss_weight
+            logging.info(f"  energy: {args.energy_loss_weight}")
+        
+        # Set forces weight (key depends on model type)
+        if args.forces_loss_weight is not None:
+            if is_conservative:
+                model.loss_weights["grad_forces"] = args.forces_loss_weight
+                logging.info(f"  grad_forces: {args.forces_loss_weight}")
+            elif is_direct:
+                model.loss_weights["forces"] = args.forces_loss_weight
+                logging.info(f"  forces: {args.forces_loss_weight}")
+        
+        # Set stress weight (key depends on model type)
+        if args.stress_loss_weight is not None:
+            if is_conservative:
+                model.loss_weights["grad_stress"] = args.stress_loss_weight
+                logging.info(f"  grad_stress: {args.stress_loss_weight}")
+            elif is_direct and "stress" in model.heads:
+                model.loss_weights["stress"] = args.stress_loss_weight
+                logging.info(f"  stress: {args.stress_loss_weight}")
+        
+        logging.info(f"Final loss_weights: {model.loss_weights}")
+        logging.info("=" * 60)
```

Comment:
Same here, I think it's cleaner to add an argument in `orb_v3_conservative_inf_omat` (and others) called `loss_weights` that takes a dictionary and passes to the constructor of the model.